### PR TITLE
implement Tick division, fix Timedelta.__cmp__ tick

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -126,6 +130,26 @@ def apply_index_wraps(func):
     return wrapper
 
 
+cdef _wrap_timedelta_result(result):
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/offsets.pyx]: eventually we'll reuse this for rdiv, floordiv, etc.

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 2]
```diff
@@ -78,6 +81,27 @@ def test_unary_ops(self):
 
 
 class TestTimedeltaComparison(object):
+    @pytest.mark.parametrize('tick_cls', [Nano, Micro, Milli, Second,
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_timedelta.py]: dont' we have this in a fixture? tick_classes?

File: pandas/_libs/tslibs/offsets.pyx [hunk 3]
```diff
@@ -126,6 +130,26 @@ def apply_index_wraps(func):
...
+    ----------
+    result : object
+
+    Returns
+    -------
+    object
+    """
+    if PyDelta_Check(result):
+        # convert Timedelta back to a Tick
+        from pandas.tseries.offsets import _delta_to_tick
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx]: this would be easy in cython, except for the fact that the offsets are in python :<

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/offsets.pyx (hunk 3 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx]: Yah that and to_offset; im pretty sure we’ll finish moving the offsets to cython eventually (may need to split this module)

File: pandas/tests/scalar/timedelta/test_timedelta.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: just pushed with this change

**Comment by jreback** [comment]
does this need a whatsnew note? IOW from a user facing perspective now you can compare offsets & TDs?

**Comment by jreback** [comment]
no you can put it there (0.24)

**Comment by jbrockmendel** [comment]
Done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jbrockmendel** [comment]
Great.  What's the policy for doing stuff during the RC?  i.e. there are some more Tick fixups I can do, or test cleanup/parameterization, or ... (or just be patient until after the release)

**Comment by jreback** [comment]
test things / cleanup are ok

**Pull request merged**